From bfadc1ddc79a01e7e999148912801767074794b3 Mon Sep 17 00:00:00 2001
From: xtex <xtex@astrafall.org>
Date: Fri, 19 Jun 2026 15:04:12 +0800
Subject: [PATCH] AOSCOS: move tooltip in about page to value label

What:
- Remove help buttons
- Move tooltips to value labels

Why:
- The old help buttons are too high
- Thus the line height of entries with a tooltip
  is obviously greater than that of those without
  a tooltip.

Signed-off-by: xtex <xtex@astrafall.org>
---
 kcms/about-distro/src/ui/main.qml | 16 +++++++++++-----
 1 file changed, 11 insertions(+), 5 deletions(-)

diff --git a/kcms/about-distro/src/ui/main.qml b/kcms/about-distro/src/ui/main.qml
index 7f2b4da15add..4cb336220a51 100644
--- a/kcms/about-distro/src/ui/main.qml
+++ b/kcms/about-distro/src/ui/main.qml
@@ -122,6 +122,17 @@ KCMUtils.SimpleKCM {
                                 event.accepted = true;
                             }
                         }
+
+                        QQC2.ToolTip {
+                            parent: valueLabel.handle
+                            visible: text.length > 0 && valueLabelHover.hovered
+                            text: entry.localizedHelp()
+                            delay: 100
+                        }
+
+                        HoverHandler {
+                            id: valueLabelHover
+                        }
                     }
 
                     Kirigami.Badge {
@@ -137,11 +148,6 @@ KCMUtils.SimpleKCM {
                         text: entry.localizedHint().text
                     }
 
-                    Kirigami.ContextualHelpButton {
-                        visible: toolTipText.length > 0
-                        toolTipText: entry.localizedHelp()
-                    }
-
                     QQC2.Button {
                         visible: hidden
                         property var dialog: null

base-commit: ccf18ae684c0c1b80766dd31a6e0f6379d32a523
-- 
2.52.0

